src/model/engines/engine_base.nit

```diff
@@ -53,5 +76,79 @@ class Engine
 	fun compile(submission: Submission): Bool is abstract
 
 	# Run `test` for `submission`
-	fun run_test(submission: Submission, test: TestCase): TestResult is abstract
+	fun run_test(submission: Submission, test: TestCase): TestResult do
+		var res = new TestResult(test, submission)
+
+		var tdir = "test{submission.results.length + 1}"
+		# We get a subdirectory (a testspace) for each test case
+		var ws = submission.workspace.as(not null)
```

Comment:
So the test server will crash if an error occurred during the workspace creation?